src/guide/ssr/build-config.md

```diff
@@ -0,0 +1,107 @@
+# Build Configuration
+
+The webpack config for an SSR project will be similar to a client-only project. If you're not familiar with configuring webpack, you can find more information in the documentation for [Vue CLI](https://cli.vuejs.org/guide/webpack.html#working-with-webpack) or [configuring Vue Loader manually](https://vue-loader.vuejs.org/guide/#manual-setup).
+
+## Key Differences with Client-Only Builds
+
+1. We need to create a [webpack manifest](https://webpack.js.org/concepts/manifest/) for our server-side code. This is a JSON file that webpack keeps to track how all the modules map to the output bundles.
+
+2. We should [externalize application dependencies](https://webpack.js.org/configuration/externals/). This makes the server build much faster and generates a smaller bundle file. When doing this, we have to exclude dependencies that need to be processed by webpack (like `.css`. or `.vue` files).
+
+3. We need to change webpack [target](https://webpack.js.org/concepts/targets/) to Node.js. This allows webpack to handle dynamic imports in a Node-appropriate fashion, and also tells `vue-loader` to emit server-oriented code when compiling Vue components.
+
+4. When building a server entry, we would need to define an environment variable to indicate we are working with SSR. It might be helpful to add a few `scripts` to the project's `package.json`:
+
+```json
+"scripts": {
+  "build:client": "vue-cli-service build --dest dist/client",
+  "build:server": "SSR=1 vue-cli-service build --dest dist/server",
+  "build": "rm -rf ./dist && npm run build:client && npm run build:server",
+}
+```
+
+## Example Configuration
+
+Below is a sample `vue.config.js` that adds SSR rendering to a Vue CLI project, but it can be adapted for any webpack build.
+
+```js
+const { WebpackManifestPlugin } = require('webpack-manifest-plugin')
+const nodeExternals = require('webpack-node-externals')
+const webpack = require('webpack')
+
+module.exports = {
+  chainWebpack: webpackConfig => {
```

Comment:
I was thinking whether it is a good idea or not to have 3 configs here?
One for common rules likes aliases, one for client only and one for server only. Both of them deriving from the base config (using `webpackMerge`).

The reasoning behind this is that using `chainWebpack` for re-configuration is hardly maintainable. It does get very complex extremely fast so noone want to touch this config because noone understands what happens here. I have the same setup on my project and it overwrites a ton of rules exactly like shown here: replacements\deletions inside `if` statements which depend on environment. I think a static webpack config is way more readable because you can understand what it does at the first glance just by the structure of that config (given you have an experience with webpack). This programmatic config is hard to read because you have to parse it line by line working as an interpreter. It is also unclear what the base vue-cli config looks like, so all the `delete` calls look like magic to me.

Reply:
I agree with the point but I have a few questions regarding it:
- should we move away from Vue CLI in this case? I am not sure how it's done correctly in the CLI environment as I usually always went with `configureWebpack` / `chainWebpack` in the `vue.config.js

Reply:
Added a follow-up to refactor to CLI agnostic build with a base, server and client configs: https://github.com/vuejs/docs-next/issues/969

Reply:
Yeah, this is probably not feasible with current vue-cli API, so it should be a part of vue-cli agnostic build in that case.